### PR TITLE
Add RETRY_DELAY option

### LINKED ISSUE #80: `DELAY` attribute specifically for retries
I assumed the `DELAY` attr would set the delay for retries but instead it applies to *all* requests. I would appreciate it if there was a `DELAY` attr specifically for retries (`RETRY_DELAY`). I'd be happy to implement it if given the go-ahead.

Thank you for this great library!

### PR DESCRIPTION
**Pull request by abmyii**
Closes #80.

### CODE REVIEW

**Comment by abmyii** [comment]
Will tests/examples/docs be necessary?

**Comment by howie6879** [comment]
Great work👍, by the way, can you add the test case and doc?

abmyii <notifications@github.com>于2020年1月1日 周三下午6:43写道：

**Comment by abmyii** [comment]
Done. I probably shouldn't have amended but did so because I wanted the default `RETRY_DELAY` to be `0`, and had already added the test.

**Comment by abmyii** [comment]
I'm not sure where to add it in the docs - any suggestions?

**Comment by howie6879** [comment]
> I'm not sure where to add it in the docs - any suggestions?

Ruia's documentation is really not very detailed, so it's a good place to add it at this point, [Request](https://docs.python-ruia.org/en/apis/request.html)

**Comment by abmyii** [comment]
OK, will do!

**Comment by howie6879** [comment]
> OK, will do!

Ruia documentation is really bad at the moment, we need a native English speaker to maintain the documentation, I do not know if you are interested in joining Ruia to do such a very important thing

**Comment by abmyii** [comment]
I'm planning to use ruia for a project, so I expect I'll need to refer back to the docs at times. Perhaps then I could fill in the docs as I go along (it will be gradual - is that OK)? Also I'm not 100% with async/coroutines so I'd need advice before writing up the docs in that regard.
Otherwise, yes!

**Comment by howie6879** [comment]
> I'm planning to use ruia for a project, so I expect I'll need to refer back to the docs at times. Perhaps then I could fill in the docs as I go along (it will be gradual - is that OK)? Also I'm not 100% with async/coroutines so I'd need advice before writing up the docs in that regard.
> Otherwise, yes!

That's a great idea. I look forward to your changes to Ruia's documentation. Thank you

**Comment by abmyii** [comment]
Oh BTW, are the CN docs more complete? If you write the CN docs, I can translate them to EN to keep them both up to date.

**Comment by abmyii** [comment]
I documented all the options in one go - I couldn't think of an easy way to split the other options into a new PR.

**Comment by abmyii** [comment]
Added default values for the options.
Also, `ruia` is a nice name. I was wondering - what does it stand for?

**Comment by howie6879** [comment]
> I documented all the options in one go - I couldn't think of an easy way to split the other options into a new PR.

Great work

**Comment by howie6879** [comment]
> Added default values for the options.
> Also, `ruia` is a nice name. I was wondering - what does it stand for?

My lover's abbreviation

**Comment by howie6879** [comment]
About project name and logo, see https://github.com/howie6879/ruia/issues/18

**Comment by abmyii** [comment]
Great, thanks for merging!

File: ruia/request.py [hunk 1]
```diff
@@ -188,6 +189,10 @@ def current_request_session(self):
     async def _retry(self, error_msg):
         """Manage request"""
         if self.retry_times > 0:
+            # Sleep to give server a chance to process/cache prior request
+            if self.request_config.get("RETRY_DELAY", 0) > 0:
+                await asyncio.sleep(self.request_config["RETRY_DELAY"])
```
[the remark this answers is not in the document]
Comment (abmyii) [reply ruia/request.py]: I think `RETRY_DELAY` should override `DELAY`, though perhaps there should be an option to combine? I'm thinking of a scenario where a site is checked every 60 mins - whenever there is a retry necessary it should do so without waiting 60 mins + `RETRY_DELAY`.

**Comment by abmyii** [comment]
If the fix is acceptable I'll update the test and docs.

File: ruia/request.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (abmyii) [reply ruia/request.py]: Sorry - I did this after PR was merged and don't know how to make it apply now:
https://github.com/abmyii/ruia/commit/f23fcccac75e77a84fbe97f85fb2bfb22488969f

If the fix is acceptable I'll update the test and docs.

**Comment by howie6879** [comment]
maybe you should submit a new PR

abmyii <notifications@github.com>于2020年1月1日 周三下午10:31写道：

**Comment by howie6879** [comment]
For the request to retry again, I also think that the delay defined for the
first time should not be considered, just use retry_reply is the best.

howie hu <howie6879@gmail.com>于2020年1月1日 周三下午10:35写道：

**Pull request merged**